hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -6423,7 +6427,11 @@ protected void restoreEdit(HStore s, Cell cell, MemStoreSizing memstoreAccountin
    * @throws IOException
    */
   private static boolean isZeroLengthThenDelete(final FileSystem fs, final FileStatus stat,
-      final Path p) throws IOException {
+      final Path p, Configuration conf) throws IOException {
+    if (stat.getLen() > 0) {
+      return false;
+    }
+    RecoverLeaseFSUtils.recoverFileLease(fs, p, conf);
```

Comment:
But after this, you do not get the stat again?
Anyway, my point here is that, make sure that when we call this method, we have already finished renaming the wal directory and calling recover lease on the wal files. Calling recoverLease and then re-get the FileStatus will generate extra load to NameNode.

Reply:
https://github.com/apache/hbase/blob/master/hbase-server/src/main/java/org/apache/hadoop/hbase/master/region/MasterRegion.java#L255
I think we can make sure we finish rename and recover lease

Reply:
OK, then we do not need to add extra recover lease here.

Reply:
I see now (was resolved so I missed it).

Then the patch looks good. A nit is that we could do w/ a comment here that says it is safe to trust the zero-length in this case because we've been through rename and lease recovery in the above. Thanks.